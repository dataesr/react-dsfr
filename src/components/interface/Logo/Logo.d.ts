import * as React from 'react';

export type LogoClassName = string | Object | any[];

export type LogoChildren = string | React.ReactNode;

export interface LogoProps {
    __TYPE?: any;
    className?: LogoClassName;
    children: LogoChildren;
    hrefTitle?: string;
    href?: string;
    /**
     * Number of characters to split title
     */
    splitCharacter?: number;
    asLink?: React.ReactElement<any>;
}

declare const Logo: React.FC<LogoProps>;

export default Logo;

