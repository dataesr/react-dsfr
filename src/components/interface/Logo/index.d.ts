import type Logo from "./Logo"

export type { Logo }
