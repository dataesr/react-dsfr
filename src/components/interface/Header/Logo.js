import classnames from 'classnames';
import PropTypes from 'prop-types';
import { CLASS_NAME_TYPE } from '../../../utils/variables';

const Logo = ({ src, alt, className }) => (
  <div className={classnames(className, 'rf-header__operator')}>
    <img src={src} alt={alt} />
  </div>
);

Logo.defaultProps = {
  className: '',
};

Logo.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  /**
 * One of: string, object
 */
  className: CLASS_NAME_TYPE,
};
export default Logo;
