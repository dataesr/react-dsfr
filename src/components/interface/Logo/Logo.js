import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import classNames from 'classnames';
import Link from '../Link/index';
import dataAttributes from '../../../utils/data-attributes';
import typeValidation from '../../../utils/type-validation';

/*
* DSFR v1.0.0
*/
import '@gouvfr/dsfr/dist/css/core.min.css';
import '@gouvfr/dsfr/dist/css/content.min.css';
import '@gouvfr/dsfr/dist/css/forms.min.css';
import '@gouvfr/dsfr/dist/css/inputs.min.css';

/*
* react-ds-fr
*/
import '../../../style/colors.css';
import '../../../style/custom.css';

import '@gouvfr/dsfr/dist/css/logo.min.css';

/**
 *
 * @visibleName Logo
 */
const Logo = ({
  children, href, to, className, splitCharacter, hrefTitle, ...remainingProps
}) => {
  // TODO Better system to split
  const regex = new RegExp(`.{${splitCharacter.toString()}}\\S*\\s+`, 'g');
  const arrayStr = children.replace(regex, '$&@').split(/\s+@/);
  const title = arrayStr.reduce((el, a, i) => el.concat(a, i < arrayStr.length - 1 ? (
    <br key={uuidv4()} />) : ''), []);

  return (
    <Link
      className={classNames(className)}
      title={hrefTitle || children}
      href={href}
      to={to}
      {...dataAttributes(remainingProps)}
    >
      <p className="fr-logo">
        {title}
      </p>
    </Link>
  );
};

Logo.defaultProps = {
  __TYPE: 'Logo',
  href: undefined,
  to: undefined,
  hrefTitle: '',
  splitCharacter: 12,
  className: '',
};

Logo.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  __TYPE: typeValidation('Logo'),
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  hrefTitle: PropTypes.string,
  href: PropTypes.string,
  to: PropTypes.string,
  /**
  * Number of characters to split title
  */
  splitCharacter: PropTypes.number,
};

export default Logo;
