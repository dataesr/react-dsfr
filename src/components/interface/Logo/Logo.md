## Logo - Bloc-marque
La structure du bloc-marque est commune à toutes les entités de l’État.

```jsx
<Logo>
    Ministère de l'enseignement supérieur de la recherche et de l'innovation
</Logo>
```
