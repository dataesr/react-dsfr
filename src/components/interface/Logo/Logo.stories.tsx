import { ComponentStory } from '@storybook/react';

import Logo from './Logo';

export default {
  title: 'Composants/Logo',
  component: Logo,
  parameters: {
    docs: {
      description: {
        component: 'Le composant permet de formatter le texte qui accompagne le logo du Ministère.',
      },
    },
  },
};

const Template: ComponentStory<typeof Logo> = (args) => <Logo {...args} />;

export const FormatLogo = Template.bind({});
FormatLogo.args = {
  children: 'Ministère de l\'enseignement supérieur et de la Recherche',
};
FormatLogo.storyName = 'Logo formatté';
