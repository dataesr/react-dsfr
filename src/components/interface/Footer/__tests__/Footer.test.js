import { render, screen } from '@testing-library/react';
import {
  FooterBody,
  FooterTop,
  FooterTopCategory,
  FooterBottom,
  FooterPartners,
  FooterLink,
  FooterCopy,
  FooterPartnersTitle,
  FooterPartnersLogo,
  Footer,
} from '..';
import Logo from '../../Logo';

describe('<Footer />', () => {
  it('should render Footer properly', () => {
    render(
      <Footer>
        <FooterTop>
          <FooterTopCategory title="Nom de la catégorie">
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
          </FooterTopCategory>
          <FooterTopCategory title="Nom de la catégorie">
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
          </FooterTopCategory>
          <FooterTopCategory title="Nom de la catégorie">
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
          </FooterTopCategory>
          <FooterTopCategory title="Nom de la catégorie">
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
          </FooterTopCategory>
          <FooterTopCategory title="Nom de la catégorie">
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
            <FooterLink href="/">Lien de navigation</FooterLink>
          </FooterTopCategory>
        </FooterTop>
        <FooterBody
          description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Uenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
        >
          <Logo splitCharacter={10}>
            Ministère de l&apos;enseignement supérieur de la rechercher et de l&apos;innovation
          </Logo>
        </FooterBody>
        <FooterPartners>
          <FooterPartnersTitle>Nos partenaires</FooterPartnersTitle>
          <FooterPartnersLogo isMain href="/" imageSrc="img/logo1" imageAlt="Logo 1" />
          <FooterPartnersLogo href="/" imageSrc="img/logo2" imageAlt="Logo 2" />
          <FooterPartnersLogo href="/" imageSrc="img/logo3" imageAlt="Logo 3" />
          <FooterPartnersLogo href="/" imageSrc="img/logo4" imageAlt="Logo 4" />
        </FooterPartners>
        <FooterBottom>
          <FooterLink href="/">Plan du site</FooterLink>
          <FooterLink href="/">Accessibilité: non/partiellement/totalement conforme</FooterLink>
          <FooterLink href="/">Mentions légales</FooterLink>
          <FooterLink href="/">Données personnelles</FooterLink>
          <FooterLink href="/">Gestion des cookies</FooterLink>
          <FooterCopy href="/">© République Française 2020</FooterCopy>
        </FooterBottom>
      </Footer>,
    );
    const footer = screen.getByTestId('footer');
    expect(footer).toMatchSnapshot();
  });
});
