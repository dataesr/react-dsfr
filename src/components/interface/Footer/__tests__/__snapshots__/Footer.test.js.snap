// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render Footer properly 1`] = `
<footer
  class="fr-footer"
  data-testid="footer"
  role="contentinfo"
>
  <div
    class="fr-footer__top"
    data-testid="footertop"
  >
    <div
      class="fr-container"
    >
      <div
        class="fr-grid-row fr-grid-row--center fr-grid-row--gutters"
      >
        <div
          class="fr-col-12 fr-col-sm-4 fr-col-md-2"
        >
          <div
            class="fr-footer__top-cat"
          >
            Nom de la catégorie
          </div>
          <ul
            class="fr-footer__top-list"
          >
            <li
              class="fr-footer__top-item"
              data-testid="footerlink"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
          </ul>
        </div>
        <div
          class="fr-col-12 fr-col-sm-4 fr-col-md-2"
        >
          <div
            class="fr-footer__top-cat"
          >
            Nom de la catégorie
          </div>
          <ul
            class="fr-footer__top-list"
          >
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
          </ul>
        </div>
        <div
          class="fr-col-12 fr-col-sm-4 fr-col-md-2"
        >
          <div
            class="fr-footer__top-cat"
          >
            Nom de la catégorie
          </div>
          <ul
            class="fr-footer__top-list"
          >
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
          </ul>
        </div>
        <div
          class="fr-col-12 fr-col-sm-4 fr-col-md-2"
        >
          <div
            class="fr-footer__top-cat"
          >
            Nom de la catégorie
          </div>
          <ul
            class="fr-footer__top-list"
          >
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
          </ul>
        </div>
        <div
          class="fr-col-12 fr-col-sm-4 fr-col-md-2"
        >
          <div
            class="fr-footer__top-cat"
          >
            Nom de la catégorie
          </div>
          <ul
            class="fr-footer__top-list"
          >
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
            <li
              class="fr-footer__top-item"
            >
              <a
                class="fr-footer__top-link"
                href="/"
                target="_self"
              >
                Lien de navigation
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    class="fr-container"
  >
    <div
      class="fr-footer__body"
      data-testid="footerbody"
    >
      <div
        class="fr-footer__brand"
      >
        <a
          class=""
          data-testid="footerlogo"
          target="_self"
          title="Ministère de l'enseignement supérieur de la rechercher et de l'innovation"
        >
          <p
            class="fr-logo"
          >
            Ministère de
            <br />
            l'enseignement
            <br />
            supérieur de
            <br />
            la rechercher
            <br />
            et de l'innovation
            
          </p>
        </a>
      </div>
      <div
        class="fr-footer__content"
      >
        <p
          class="fr-footer__content-desc"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Uenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>
        <ul
          class="fr-footer__content-list"
        >
          <li
            class="fr-footer__content-item"
          >
            <a
              class="fr-footer__content-link"
              href="https://legifrance.gouv.fr"
              target="_self"
            >
              legifrance.gouv.fr
            </a>
          </li>
          <li
            class="fr-footer__content-item"
          >
            <a
              class="fr-footer__content-link"
              href="https://gouvernement.fr"
              target="_self"
            >
              gouvernement.fr
            </a>
          </li>
          <li
            class="fr-footer__content-item"
          >
            <a
              class="fr-footer__content-link"
              href="https://service-public.fr"
              target="_self"
            >
              service-public.fr
            </a>
          </li>
          <li
            class="fr-footer__content-item"
          >
            <a
              class="fr-footer__content-link"
              href="https://data.gouv.fr"
              target="_self"
            >
              data.gouv.fr
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="fr-footer__partners"
      data-testid="footerpartners"
    >
      <p
        class="fr-footer__partners-title"
        data-testid="title"
      >
        Nos partenaires
      </p>
      <div
        class="fr-footer__partners-logos"
      >
        <div
          class="fr-footer__partners-main"
        >
          <a
            class="footer__partners-link"
            data-testid="logo"
            href="/"
            target="_self"
          >
            <img
              alt="Logo 1"
              class="fr-footer__logo"
              src="img/logo1"
            />
          </a>
        </div>
        <div
          class="fr-footer__partners-sub"
        >
          <a
            class="footer__partners-link"
            href="/"
            target="_self"
          >
            <img
              alt="Logo 2"
              class="fr-footer__logo"
              src="img/logo2"
            />
          </a>
          <a
            class="footer__partners-link"
            href="/"
            target="_self"
          >
            <img
              alt="Logo 3"
              class="fr-footer__logo"
              src="img/logo3"
            />
          </a>
          <a
            class="footer__partners-link"
            href="/"
            target="_self"
          >
            <img
              alt="Logo 4"
              class="fr-footer__logo"
              src="img/logo4"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="fr-footer__bottom"
      data-testid="bottom"
    >
      <ul
        class="fr-footer__bottom-list"
      >
        <li
          class="fr-footer__bottom-item"
          data-testid="link"
        >
          <a
            class="fr-footer__bottom-link"
            href="/"
            target="_self"
          >
            Plan du site
          </a>
        </li>
        <li
          class="fr-footer__bottom-item"
        >
          <a
            class="fr-footer__bottom-link"
            href="/"
            target="_self"
          >
            Accessibilité: non/partiellement/totalement conforme
          </a>
        </li>
        <li
          class="fr-footer__bottom-item"
        >
          <a
            class="fr-footer__bottom-link"
            href="/"
            target="_self"
          >
            Mentions légales
          </a>
        </li>
        <li
          class="fr-footer__bottom-item"
        >
          <a
            class="fr-footer__bottom-link"
            href="/"
            target="_self"
          >
            Données personnelles
          </a>
        </li>
        <li
          class="fr-footer__bottom-item"
        >
          <a
            class="fr-footer__bottom-link"
            href="/"
            target="_self"
          >
            Gestion des cookies
          </a>
        </li>
      </ul>
      <div
        class="fr-footer__bottom-copy"
      >
        © République Française 2020
      </div>
    </div>
  </div>
</footer>
`;
